system/Router/DefinedRouteCollector.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Router;
+
+use Closure;
+use Generator;
+
+/**
+ * Collect all defined routes for display.
+ */
+class DefinedRouteCollector
+{
+    private RouteCollection $routeCollection;
+
+    public function __construct(RouteCollection $routes)
+    {
+        $this->routeCollection = $routes;
+    }
+
+    public function collect(): Generator
```

Comment:
Can you add the signature of the Generator in the PHPDoc so that it can be used by IDEs?

Reply:
Done.

Reply:
That sig begs the question: is it worth introducing a value object at some point for better definition?

Reply:
The data structure of a route is complex, so it would be easier to understand if it were a class.